### PR TITLE
fix Arbitrage list issue.

### PR DESCRIPTION
**Pull request by rick23p**

### CODE REVIEW

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -206,3 +206,7 @@ export const abbreviateNumber = (
   }
   return shortValue + suffixes[suffixNum];
 };
+
+export const isEmpty = val => {
+  return val === undefined || val == null ? true : false;
```
Comment (tiltom) [inline src/utils/helpers.ts:211]:
```suggestion
   val === undefined || val == null;
```

File: src/utils/helpers.ts [hunk 2]
```diff
@@ -206,3 +206,7 @@ export const abbreviateNumber = (
   }
   return shortValue + suffixes[suffixNum];
 };
+
+export const isEmpty = val => {
```
Comment (tiltom) [inline src/utils/helpers.ts:210]:
```suggestion
export const isNullOrUndefined = val => {
```

File: src/app/components/Arbitrage/Arbitrage.tsx [hunk 3]
```diff
@@ -31,7 +31,16 @@ export function Arbitrage() {
   const opportunityArray = useMemo(
     () =>
       Object.values(data)
-        .filter(item => item.hasOwnProperty('rateToBalance'))
+        .filter(item => {
+          const rateToBalance = item?.rateToBalance;
```
Comment (tiltom) [inline src/app/components/Arbitrage/Arbitrage.tsx:35]: You can extract this whole check to a local function in this file (e.g. `isValidArbitrage`).

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/utils/helpers.ts [hunk 4]
```diff
@@ -207,6 +207,6 @@ export const abbreviateNumber = (
   return shortValue + suffixes[suffixNum];
 };
 
-export const isEmpty = val => {
-  return val === undefined || val == null ? true : false;
+export const isNullOrUndefined = val => {
```
Comment (tiltom) [inline src/utils/helpers.ts:210]: You have unnecessary curly braces and `return` here.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/utils/helpers.ts [hunk 5]
```diff
@@ -206,3 +206,5 @@ export const abbreviateNumber = (
   }
   return shortValue + suffixes[suffixNum];
 };
+
+export const isNullOrUndefined = val => val === undefined || val == null;
```
Comment (ShynRou) [inline src/utils/helpers.ts:210]: `val == null` already checks for null and undefined.

In my opinion this function is redundant, you could just check `rateToBalance.earn != null`.

Comment (tiltom) [reply src/utils/helpers.ts:210]: Usage of `==` in JS is antipattern and I wouldn't use it at all.

Comment (tiltom) [reply src/utils/helpers.ts:210]: And I haven't noticed it's used here, good catch :D @rick23p please change `val == null` to `vall === null`.

**Review by creed-victor — APPROVED** [review]

**Review by tiltom — APPROVED** [review]

**Pull request merged**